clipper_admin/clipper_admin/docker/docker_container_manager.py

```diff
@@ -237,8 +241,17 @@ def set_num_replicas(self, name, version, input_type, image, num_replicas):
                     name=name,
                     version=version,
                     missing=(num_missing)))
+
+            model_container_names = []
             for _ in range(num_missing):
-                self._add_replica(name, version, input_type, image)
+                name = self._add_replica(name, version, input_type, image)
+                model_container_names += [name]
```

Comment:
Change this to `model_container_names.append(name)`. The way you're doing it right now constructs a new list in each iteration.